### DISCUSSION TITLE
How to get count of non existing values from eloquent?

### DISCUSSION THREAD

**Discussion by ruvaifa-masood** [laravel/framework#38291] [Q&A]
Hi there,

Is there any way to get count of non existing values from eloquent?

For example:

```php
$count = Post::whereBetween('created_at', ['from date', 'to date'])->count();
```

How can I get the count of those posts which doesn't exist between the mentioned from and to dates?

**Comment by dennisprudlo** [comment]
Do you mean all posts that weren't created in the specified range? Like so?

```php
$count = Post::where('created_at', '<', $fromDate)->where('created_at', '>', $toDate)->count();
```

**Reply by ruvaifa-masood** [reply]
Yes, but there should be a method in eloquent to get the count of non existing values.

**Reply by hotmeteor** [reply]
Wouldn't you just get a count of dates for that range and then subtract the number that exist?

**Reply by ruvaifa-masood** [reply]
It may cause some performance issues. I think it would be better if it calculates within the database layer.

**Reply by dennisprudlo** [reply]
Instead of `whereBetween` you could use `whereNotBetween`.
